core/src/main/java/org/apache/gravitino/connector/BaseCatalog.java

```diff
@@ -181,53 +182,64 @@ public CatalogOperations ops() {
 
   public AuthorizationPlugin getAuthorizationPlugin() {
     if (authorization == null) {
-      synchronized (this) {
-        if (authorization == null) {
-          BaseAuthorization<?> baseAuthorization = createAuthorizationPluginInstance();
-          if (baseAuthorization == null) {
-            return null;
-          }
-          authorization = baseAuthorization;
-        }
-      }
+      return null;
     }
     return authorization.plugin(provider(), this.conf);
   }
 
-  private BaseAuthorization<?> createAuthorizationPluginInstance() {
+  public void initAuthorizationPluginInstance(IsolatedClassLoader classLoader) {
+    if (authorization != null) {
+      return;
+    }
+
     String authorizationProvider =
         catalogPropertiesMetadata().containsProperty(AUTHORIZATION_PROVIDER)
             ? (String) catalogPropertiesMetadata().getOrDefault(conf, AUTHORIZATION_PROVIDER)
             : null;
 
     if (authorizationProvider == null) {
       LOG.info("Authorization provider is not set!");
-      return null;
+      return;
```

Comment:
Do we only add logs if the authorization provider is not set?

Reply:
Yes, the Authorization plugin is an option in the Catalog. It may or may not be configured.